packages/circuit-ui/components/ModalContext/ModalContext.spec.tsx

```diff
@@ -33,25 +35,15 @@ const Modal: ModalComponent = ({ onClose }) => (
 Modal.TRANSITION_DURATION = 200;
 
 describe('ModalContext', () => {
-  beforeAll(() => {
-    jest.useFakeTimers();
-  });
   afterAll(() => {
-    jest.useRealTimers();
-    jest.resetModules();
+    vi.resetModules();
   });
   afterEach(() => {
-    jest.clearAllMocks();
+    vi.clearAllMocks();
   });
 
-  /**
-   * We need to set up userEvent with delay=null to address this issue:
-   * https://github.com/testing-library/user-event/issues/833
-   */
-  const userEvent = baseUserEvent.setup({ delay: null });
```

Comment:
Vite's fake timers seem to be similarly incompatible with `userEvent`. However, this workaround (and others) don't fix it. I've removed the fake timers at the cost of a couple of hundred milliseconds of run time.

Reply:
What about if we want to use fake timers in more places in the future? Would we just not do it?

Is there any WIP on this from the vitest and/or testing-library teams, issues we could follow?

Reply:
Fake timers are still used in a couple of test suites. The incompatibility is only with `userEvent`. I didn't dig too deep since I felt that the tradeoff was acceptable, though you're right that we should investigate this in/for the future.

Reply:
I did more research and found two related issues: 

- https://github.com/testing-library/react-testing-library/issues/1197
- https://github.com/testing-library/react-testing-library/issues/1187

The first one suggests a temporary workaround that works for our use case, so I've reverted the fake timer changes in 8134d02.

Reply:
Thank you for looking into it!!